src/Features/Core/Portable/Wrapping/InitializerExpression/InitializerExpressionCodeActionComputer.cs

```diff
@@ -0,0 +1,277 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.CodeAnalysis.Wrapping.SeparatedSyntaxList;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Wrapping.InitializerExpression
+{
+    internal abstract partial class AbstractInitializerExpressionWrapper<TListSyntax, TListItemSyntax>
+    {
+        private class InitializerExpressionCodeActionComputer : AbstractCodeActionComputer<AbstractInitializerExpressionWrapper<TListSyntax, TListItemSyntax>>
+        {
+            private readonly TListSyntax _listSyntax;
+            private readonly SeparatedSyntaxList<TListItemSyntax> _listItems;
+            private readonly SyntaxTrivia _afterOpenTokenIndentationTrivia;
+            private readonly SyntaxTrivia _singleIndentationTrivia;
+            private readonly SyntaxTrivia _braceIndentationTrivia;
+            private readonly bool _doWrapInitializerOpenBrace;
+
+            public InitializerExpressionCodeActionComputer(
+                AbstractInitializerExpressionWrapper<TListSyntax, TListItemSyntax> service,
+                Document document, SourceText sourceText, DocumentOptionSet options,
+                TListSyntax listSyntax, SeparatedSyntaxList<TListItemSyntax> listItems,
+                bool doWrapInitializerOpenBrace, CancellationToken cancellationToken)
+                : base(service, document, sourceText, options, cancellationToken)
+            {
+                _listSyntax = listSyntax;
+                _listItems = listItems;
+
+                var generator = SyntaxGenerator.GetGenerator(OriginalDocument);
+
+                _afterOpenTokenIndentationTrivia = generator.Whitespace(GetAfterOpenTokenIdentation());
+                _singleIndentationTrivia = generator.Whitespace(GetSingleIdentation());
+                _braceIndentationTrivia = generator.Whitespace(GetBraceTokenIndentation());
+
+                _doWrapInitializerOpenBrace = doWrapInitializerOpenBrace;
+            }
+
+            private string GetAfterOpenTokenIdentation()
+            {
+                var openToken = _listSyntax.GetFirstToken();
+                var afterOpenTokenOffset = OriginalSourceText.GetOffset(openToken.Span.End);
+
+                var indentString = afterOpenTokenOffset.CreateIndentationString(UseTabs, TabSize);
+                return indentString;
+            }
+
+            private string GetBraceTokenIndentation()
+            {
+                var previousToken = _listSyntax.GetFirstToken().GetPreviousToken();
```

Comment:
is this not the same as `_listSyntax.GetFirstToken()`?

Reply:
No, the listSyntax contains the open brace, elements, and end brace. I want to get the token before that to calculate the appropriate indentation for the whole listSyntax.